### PR TITLE
Added SNIPES presence

### PR DESCRIPTION
**Pull request by N0chteil**
Proof:
Screenshot_289
Screenshot_290
Screenshot_291

### CODE REVIEW

File: websites/S/Snipes/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.0",
```
Comment (StrikerFRFX) [inline websites/S/Snipes/dist/metadata.json:2]: Use latest schema

File: websites/S/Snipes/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "www.snipes.nl",
+    "www.snipes.it",
+    "www.snipes.es"
+  ],
+  "version": "1.0.0",
+  "logo": "https://i.imgur.com/9HmTAOR.png",
+  "thumbnail": "https://i.imgur.com/bKZ2EQH.png",
+  "color": "#DF632F",
+  "tags": [
+    "snipes",
```
Comment (StrikerFRFX) [inline websites/S/Snipes/dist/metadata.json:25]: Remove website name from tags as not an alternate service name

File: websites/S/Snipes/presence.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  const num = (urlpath[2] === 'men' || urlpath[2] === 'women' || urlpath[2] === 'kids') ? "3" : "2";
+  if(document.querySelector("a.b-refinements-category-link.active")){
+    const category = document.querySelector("a.b-refinements-category-link.active").getAttribute("data-name");
+    presenceData.state = (urlpath[parseInt(num) + 1]) ? category : "";
+  }
+  if(set_showButtons) {
+    presenceData.buttons = [
+      {
+        label: 'Show Category',
+        url: window.location.href
```
Comment (StrikerFRFX) [inline websites/S/Snipes/presence.ts:29]: Please provide a reference to which URL (or part of website) this button will redirect to

Comment (StrikerFRFX) [reply websites/S/Snipes/presence.ts:29]: Also wouldn't `View Category` be more appropriate as a label for this?

File: websites/S/Snipes/presence.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  }
+} else if (urlpath[1] === "p") {
+  const prod = document.querySelector("div.js-target").textContent,
+        prodModel = document.querySelector("div.js-target>.h-hide").textContent,
+        brand = document.querySelector("div.js-target>a").textContent,
+        product = prod.replace(prodModel, "").replace(brand, "").replace(/\s+/g, " ").trim();
+  if(set_showButtons) {
+    presenceData.buttons = [
+      {
+        label: 'Show Product',
```
Comment (StrikerFRFX) [inline websites/S/Snipes/presence.ts:65]: Same here as other button

File: websites/S/Snipes/presence.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+} else if (urlpath[1].startsWith("search")) {
+  const urlParams = new URLSearchParams(window.location.search);
+  presenceData.details = "Search:";
+  presenceData.state = urlParams.get('q');
+
+  if(set_showButtons) {
+    presenceData.buttons = [
+      {
+        label: 'Show Results',
+        url: window.location.href
```
Comment (StrikerFRFX) [inline websites/S/Snipes/presence.ts:82]: Same here

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
Hi @N0chteil, 
I've requested changes below in my review which you should take a look at 😄 

Thanks,
**Striker**, PreMiD Reviewer Team.
Contact: me@strikerfrfx.dev, Striker#1337

File: websites/S/Snipes/dist/metadata.json (hunk 1 above)
Comment (ririxidev) [reply websites/S/Snipes/dist/metadata.json]: It doesn't matter bruh. $schema is only for client-side validation for example if they have plugin in vscode to validate schema.

Comment (StrikerFRFX) [reply websites/S/Snipes/dist/metadata.json]: Yes but if they're going to use schema they might as well use the latest one or not use it at all, simple.

Comment (ririxidev) [reply websites/S/Snipes/dist/metadata.json]: 🤦

**Review by Bas950 — DISMISSED** [review]
LGTM

File: websites/S/Snipes/presence.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  const presenceData = {
+        largeImageKey: "logo"
+      },
+      set_timeElapsed = await presence.getSetting("timeElapsed"),
+      set_showButtons = await presence.getSetting("showButtons"),
+      urlpath = window.location.pathname.split("/");
+
+  if (set_timeElapsed) {
+    presenceData.startTimestamp = browsingStamp;
+  }
```
Comment (ririxidev) [inline websites/S/Snipes/presence.ts:16]: You don't need brackets here.

File: websites/S/Snipes/presence.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  if(!urlpath[1]) {
+    presenceData.details = "Home";
+  } else if (urlpath[1] === "c") {
```
Comment (ririxidev) [inline websites/S/Snipes/presence.ts:19]: same here like above

File: websites/S/Snipes/presence.ts [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+    const num = (urlpath[2] === 'men' || urlpath[2] === 'women' || urlpath[2] === 'kids') ? "3" : "2";
+    if(document.querySelector("a.b-refinements-category-link.active")){
+      const category = document.querySelector("a.b-refinements-category-link.active").getAttribute("data-name");
+      presenceData.state = (urlpath[parseInt(num) + 1]) ? category : "";
```
Comment (ririxidev) [inline websites/S/Snipes/presence.ts:23]: Please don't set state to "" nor undefined nor null.
```suggestion
      const category = document.querySelector("a.b-refinements-category-link.active").getAttribute("data-name");
      if (urlpath[parseInt(num) + 1]) presenceData.state = category;
```

File: websites/S/Snipes/presence.ts [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    } else if(urlpath[2] === 'sale') {
+      presenceData.details = "Sale";
+    } else if(urlpath[2] === 'deals') {
+      presenceData.details = "Deals";
+    } else if(urlpath[2] === 'musthaves') {
+      presenceData.details = "Must haves";
+    } else if(document.querySelector("li.b-breadcrumb-item>span.b-breadcrumb-text")) {
+      presenceData.details = "Category:";
+      presenceData.state = document.querySelector("li.b-breadcrumb-item>span.b-breadcrumb-text").textContent;
+    }
```
Comment (ririxidev) [inline websites/S/Snipes/presence.ts:56]: You don't need brackets for one line code in if

File: websites/S/Snipes/presence.ts [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    presenceData.details = "Newsletter settings";
+  } else if (urlpath[1] === "login") {
+    presenceData.details = "Login";
+  } else if (urlpath[1] === "registration") {
+      presenceData.details = "Register";
+  } else if (urlpath[1] === "storefinder") {
+    presenceData.details = "Storefinder";
+  } else if (urlpath[1] === "content") {
+    presenceData.details = "Other";
+  }
```
Comment (ririxidev) [inline websites/S/Snipes/presence.ts:114]: Same like above

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @N0chteil!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

**Review by ririxidev — APPROVED** [review]
lgtm

**Review by StrikerFRFX — APPROVED** [review]
lgtm

**Pull request merged**